### PR TITLE
drivers/lpsxxx: add SPI bus support

### PR DESCRIPTION
**Pull request by wingunder**
# Contribution description

Based on issue #13208, this PR adds SPI support to the lpsxxx driver. The driver now supports I2C as well as SPI, but the unique thing is that both SPI and I2C can be supported within a single build.

The periph/comms_bus.h and periph/comms_bus.c implementation supplies the glue for pivoting between I2C and SPI busses, during run-time. These files are meant to be generic and free to be used by other peripherals that support both I2C and SPI.

### Testing procedure
A compilation-only test could be done like this:
```sh
$ cd tests/driver_lpsxxx
$ CONFIG_LPSXXX_I2C=1 make -j12 clean all
$ CONFIG_LPSXXX_SPI=1 make -j12 clean all
$ CONFIG_LPSXXX_I2C=1 CONFIG_LPSXXX_SPI=1 make -j12 clean all
```
The LPSXXX_PARAMS define could be set up to match I2C or SPI bus configurations in the following fashion.
Just be sure to include lpsxxx_params.h _after_ the following define.

```c
#define LPSXXX_PARAMS                                    \
    {                                                    \
        .transport = {                                   \
            .type = SPI,                                 \
            .bus = {                                     \
                .spi = {                                 \
                    .dev  = SPI_DEV(1),                  \
                    .cs   = GPIO_PIN(PORT_B, 12),        \
                    .clk  = (spi_clk_t)(SPI_CLK_10MHZ),  \
                    .mode = (spi_mode_t)(SPI_MODE_0),    \
                },                                       \
            },                                           \
        },                                               \
        .rate = LPSXXX_PARAM_RATE                        \
    }
```
or
```c
#define LPSXXX_PARAMS                                    \
    {                                                    \
        .transport = {                                   \
            .type = I2C,                                 \
            .bus = {                                     \
                .i2c = {                                 \
                    .dev  = I2C_DEV(0),                  \
                    .addr = 0x5d,                        \
                },                                       \
            },                                           \
        },                                               \
        .rate = LPSXXX_PARAM_RATE                        \
    }
```
The 2 parameter setups above, was successfully tested on an stm32f407 board.

### Issues/PRs references
Based on issue #13208.

### CODE REVIEW

**Comment by wingunder** [comment]
I will add the missing documentation as soon as I know that this PR is going in the right direction.

File: drivers/include/periph/comms_bus.h [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+ * @file
+ * @brief       Low-level I2C/SPI bus driver interface definition
+ *
+ * @author      Pieter du Preez <pdupreez@gmail.com>
+ */
+
+#ifndef PERIPH_COMMS_BUS_H
+#define PERIPH_COMMS_BUS_H
+
+#if !defined(MODULE_PERIPH_I2C) && !defined(MODULE_PERIPH_SPI)
```
Comment (wingunder) [inline drivers/include/periph/comms_bus.h]: This check breaks the CI build for almost all boards.
One option for fixing it, could be to simply remove this check.
I'm not sure what the best way would be to fix this?

File: drivers/include/periph/comms_bus.h [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+#error At least one of the following modules are required: periph_i2c, periph_spi
+#endif
```
Comment (wingunder) [inline drivers/include/periph/comms_bus.h]: The check above, breaks the CI build for almost all boards.
One option for fixing it, could be to simply remove the check.
I'm not sure what the best way would be to fix this?

**Comment by wingunder** [comment]
> 
> 
> I haven't look at this in detail yet but the integration into the build system should be improved (and will help with Murdock).

@aabadie Thanks. Sure, I am working on the integration.

File: drivers/include/periph/comms_bus.h [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+/**
+ * @brief   Supported bus types.
+ */
+enum comms_bus_type {
+    UNDEF = 0,
+    I2C,
+    SPI,
+};
```
Comment (wingunder) [inline drivers/include/periph/comms_bus.h]: The naming of `comms_bus_type_t`  members are way too common and clashes with defines in other projects.
It breaks the CI build.

Comment (aabadie) [reply drivers/include/periph/comms_bus.h]: You could namespace them with `COMMS_BUS_`. Example: `COMMS_BUS_I2C`. Why use a use `s` at the end of comms by the way ? I think `comm` is already ok.

Comment (aabadie) [reply drivers/include/periph/comms_bus.h]: or maybe even `common` ?

Comment (wingunder) [reply drivers/include/periph/comms_bus.h]: @aabadie: Thanks for your input. I had PERIPH_COMMS_BUS in mind. Will this be OK?

I chose `comms_bus` as an abbreviation for `communications bus`. I'm really open for anything. Let me rather change it now than later, if needed.

Would `common_bus` be better?

Comment (wingunder) [reply drivers/include/periph/comms_bus.h]: @aabadie: I followed your advice and renamed `comms_bus` to `common_bus`.
I also prefixed the `common_bus_type_t` members with `COMMON_BUS_`.
The documentation is updated to the point where doxygen is happy and the CI is passing.

Is there anything else that you would like me to do for this PR?

File: drivers/include/periph/comms_bus.h (hunk 2 above)
Comment (wingunder) [reply drivers/include/periph/comms_bus.h]: The check is not needed and was removed.

**Pull request closed without merging**